solver/llbsolver/vertex.go

```diff
@@ -69,21 +69,27 @@ func WithCacheSources(cms []solver.CacheManager) LoadOpt {
 
 func RuntimePlatforms(p []specs.Platform) LoadOpt {
 	var defaultPlatform *pb.Platform
+	var normalizedPlatform specs.Platform
 	pp := make([]specs.Platform, len(p))
 	for i := range p {
 		pp[i] = platforms.Normalize(p[i])
 	}
 	return func(op *pb.Op, _ *pb.OpMetadata, opt *solver.VertexOptions) error {
 		if op.Platform == nil {
 			if defaultPlatform == nil {
-				p := platforms.DefaultSpec()
-				defaultPlatform = &pb.Platform{
-					OS:           p.OS,
-					Architecture: p.Architecture,
-				}
+				normalizedPlatform = platforms.Normalize(platforms.DefaultSpec())
```

Comment:
if `DefaultSpec` is not already normalized for some case it should be fixed in upstream as well

Reply:
As my comments above, it is not an upstream issue, it is how we treat those variant in this project.

Reply:
We should compare normalized values, that part is correct. But `DefaultSpec` should just return normalized value not asking users to call normalize on it. I haven't checked if this is the case. If it is not it should, if it is then we don't really need the reconvert here.

Reply:
I kinds of agree with you,  both of methods come from 'containerd' project, `Normalize` did quite a lot of stuff for different platforms, cause this is also used widely in this project, this reconvert is necessary here, pls kindly see, 
https://github.com/moby/buildkit/blob/master/solver/llbsolver/vertex.go#L74

Reply:
So what is the case where the result from defaultspec changes after you call normalize on it?

Reply:
I don't know, my understanding is this will happen, but it is normal case I always face, we call some APIs from upstream/libs, we cannot guarantee those API won't change forever, so we need to patch the downstream to fit with the change, and general speaking, if this is an opened API, the upstream should notify the change whenever this happened. We cannot forecast what will happen in the upstream, so it's hard for us to provide some compatibility solutions.

The bottom line I think is we should make it works in this project.

Reply:
and since both of the methods comes from the same project (`defaultspec` and `Normalize`),  `Normalize` should take care of it if `defaultspec` has some changes. If downstream projects should use `Normalize` consistently, I don't think this is big deal.